File: src/main/java/com/evieclient/examples/ExampleModule.java


package com.evieclient.examples;

import com.evieclient.Evie;
import com.evieclient.events.bus.EventSubscriber;
import com.evieclient.modules.Category;
import com.evieclient.modules.Module;
import com.evieclient.events.impl.client.ChatReceivedEvent;

public class ExampleModule extends Module {
    public ExampleModule() { super("ExampleModule", "Example Module", Category.UTIL, true); }

    @EventSubscriber
    public void onChat (ChatReceivedEvent event) {
        Evie.log("RECEIVED CHAT! : " + event.getChatComponent().getUnformattedText());
    }
}

File: src/main/java/com/evieclient/examples/ExampleModule.kt
package com.evieclient.examples

import com.evieclient.Evie
import com.evieclient.events.bus.EventSubscriber
import com.evieclient.events.impl.client.ChatReceivedEvent
import com.evieclient.modules.Category
import com.evieclient.modules.Module

class ExampleModule : Module("ExampleModule", "Example Module", Category.UTIL, true) {
    @EventSubscriber
    fun onChat(event: ChatReceivedEvent) {
        Evie.log("RECEIVED CHAT! : " + event.chatComponent.unformattedText)
    }
}